### PR TITLE
Creating common components for fetching info

### PR DESCRIPTION
**Pull request by rcannood**
New functionality:

* `get_api_info`: extract api info from tasks

* `get_method_info`: extract method info vrom config yaml

* `get_metric_info`: extract metric info vrom config yaml

* `check_migration_status`: compare git shas from methods with v1

* `get_results`: extract benchmark scores

### CODE REVIEW

File: src/common/check_migration_status/script.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                        if obj['sha'] != comp_item['v1_commit']:
+                            output[comp_item['namespace']][comp_item['id']] = {'v1_url': comp_item['v1_url'], 'status': "not latest commit"}
+                        else :
+                            output[comp_item['namespace']][comp_item['id']] = {'v1_url': comp_item['v1_url'],'status': "up to date"}
+            else:
+                output[comp_item['namespace']][comp_item['id']] ={'v1_url': "v1_url missing"}
+
+
+with open(par['output'], 'w') as outf:
+    json.dump(output, outf)
```
Comment (rcannood) [inline src/common/check_migration_status/script.py:35]: Could you add some comments to this code w.r.t. what's going on?

You can simplify a bit by doing something like:

```python
def check_status(comp_item, git_objects) -> str:
    v1_url = comp_item["v1_url"]
    if not v1_url:
        return "v1_url missing"
```

...

...

Actually give me a second, I'll rewrite the script. @KaiWaldrant Could you review the upcoming commit?

Comment (rcannood) [reply src/common/check_migration_status/script.py]: Afaik this issue has been resolved. @KaiWaldrant Could you review the commits I added to this PR?

File: src/common/check_migration_status/script.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+output = {}
+
+with open(par['git_sha'], 'r') as f1:
+    git = json.load(f1)
+
+    with open(par['comp_info'], 'r') as f2:
+        comp = json.load(f2)
+        for comp_item in comp:
+            if comp_item['namespace'] not in output:
+                output[comp_item['namespace']] = {}
```
[the remark this answers is not in the document]
Comment (rcannood) [reply src/common/check_migration_status/script.py]: This issue has been resolved, see comment below.

File: src/common/check_migration_status/test.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import subprocess
+from os import path
+import json
+
+input_sha = meta["resources_dir"] + "resources_test/common/input_git_sha.json"
+input_method_info = meta["resources_dir"] +  "resources_test/common/method_info.json"
```
[the remark this answers is not in the document]
Comment (KaiWaldrant) [reply src/common/check_migration_status/test.py]: @rcannood created the task_metadata.sh

File: src/common/check_migration_status/script.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+## VIASH START
+par = {
+    'git_sha': 'temp/openproblems-v1.json',
+    'comp_info': 'temp/denoising_metrics.json',
+    'output': 'temp/migration_status.json'
+}
+## VIASH END
+
+def check_status(comp_item: List[Dict[str, str]], git_objects: List[Dict[str, str]]) -> str:
```
Comment (KaiWaldrant) [inline src/common/check_migration_status/script.py:12]: add List to comp_item. gave errors otherwise

Comment (KaiWaldrant) [reply src/common/check_migration_status/script.py:12]: @rcannood maybe check if it still works for you ?

**Pull request merged**